### ISSUE TITLE
OSQP erroneously diverges on QP

### ISSUE THREAD

**Issue by apuranik1** [cvxpy/cvxpy#1489]
**Describe the bug**
OSQP returns -inf on a QP problem where SCS and ECOS succeed.

**To Reproduce**
```
import cvxpy as cp
import numpy as np

n_vars = 10
ata = np.diag(np.arange(n_vars) + 1)
atb = np.array([4.0, 2.0, -2.0, 3.0, 3.0, 7.0, 3.0, 11.0, 10.0, 14.0])
n_rows = 200000
# setting n_rows = 1 makes the issue go away

var = cp.Variable(n_vars)
obj = (0.5 * cp.quad_form(var, ata) - atb.T @ var) / n_rows

constraints = []
for i in range(n_vars - 1):
    constraints.append(var[i] <= var[i + 1])

prob = cp.Problem(cp.Minimize(obj), constraints)

print(prob.solve(solver="OSQP"))
# prints -inf
print(prob.solve(solver="SCS"))
# prints -0.0001497503786444586
print(prob.solve(solver="ECOS"))
# prints -0.00014975023707310215
```

**Expected behavior**
OSQP, which I believe is also the default solver, should either produce a solution similar to the one SCS and ECOS agree on or fail explicitly rather than silently diverging.

**Output**
```
-inf
-0.0001497503786444586
-0.00014975023707310215
```

**Version**
 - OS: Arch Linux
 - CVXPY Version: 1.1.15

**Additional context**
- OSQP version: 0.6.2
Scaling up the objective function by a factor of 200000 makes it converge.
It's possible that this is the wrong place to report this bug, and this should be reported to OSQP directly.

**Comment by phschiele** [comment]
@apuranik1 Thank you for making the issue easy to reproduce. The issue appears to be solved by:

```py
prob.solve(solver="OSQP", eps_dual_inf=1e-6)
>>> -0.00014974911146097575
```

**Comment by bstellato** [comment]
Thanks for reporting this. @gbanjac @imciner2 I think this is related to the slightly different infeasibility conditions in 0.6.2. Similar issue was reported in `https://github.com/osqp/OSQP.jl/issues/94`. We should either switch back to the previous criteria or tighten these tolerances to something like `1e-06` as @phschiele suggested.

**Comment by rileyjmurray** [comment]
@bstellato @phschiele would you like us to make changes on the default tolerances for the 1.2 release?

**Comment by bstellato** [comment]
For now it might be a good fix. We already fixed in the [develop-1.0](https://github.com/osqp/osqp/tree/develop-1.0) branch but it will likely be released after cvxpy 1.2.
